src/msm.rs

```diff
@@ -30,18 +30,16 @@ impl VariableBaseMSM {
         }
     }
 
-    fn msm_slice<P: Parameters>(scalar: BigInt<P>, slices: &mut [u32], window_bits: u32) {
+    fn msm_slice<P: Parameters>(mut scalar: BigInt<P>, slices: &mut [u32], window_bits: u32) {
```

Comment:
This changes the semantics of this function signature since we don't expect msm_slice to modify the scalar. That's the reason we used a temp variable. Thoughts? @Gigatron @cyphersnake

Reply:
You take ownership of this `scalar`, it is implicitly copied on transfer to function

The semantics of the function does not change if ownership is transferred, this may be important with `&` and `&mut` but not here. Moreover, the previous function at the asm level did not change in any way, because you have already taken ownership on `scalar` and given another name to this ownership (`temp`).

Reply:
In other words, when you transfer ownership, it may not be `mut` outside, but whatever inside, this is the responsibility of the function. Implicit copying in Rust goes through the `Copy` trait and is usually implemented for types that do not leave the stack when copying

Reply:
I see. The scalar is still passed by value, so modifications inside the function do not change the value of the scalar outside.

Thanks for the clarification.